### DISCUSSION TITLE
WARNING: Minimum density

### DISCUSSION THREAD

**Discussion by ESH-y** [firemodels/fds#12676] [Q&A]
Hi,

My simulation continued till 11.2 sec then FDS showed "Warning minimum density, 0.120 kg/m3, clipped in Mesh 1" and the simulation stopped because of numerical instability. 
I wonder what is causing it.... I see it is happening in many deferent project.
any idea how can I solve it please?

**Comment by drjfloyd** [comment]
Cannot diagnose without an input file. But before attaching an input file to this discussion:

-Check to make sure you are not removing more mass than is being added to a sealed volume
-Read the User's Guide section on numerical instabilities and follow its guidance 
-Make sure you are using the current release

**Reply by ESH-y** [reply]
Hi, thank you very much for your reply.
I apologize, but unfortunately I cannot attach a file because the projects I am working on are of architects' and there is a matter of copyright.

I would appreciate it if you could explain please what can cause mass to be removed from the model? Is it meant to make sure there are no holes?

Another thing I noticed recently in the new version: FDS-6.8.0-0-g886e009-release
This warning appears:
WARNING: SPEC POLYURETHANE_REAC_fuel is not in the table of pre-defined species. Any unassigned SPEC variables in the input were assigned the properties of nitrogen.

Could it be related?


Would appreciate help
I'm really lost

**Comment by drjfloyd** [comment]
There is little that can be done here without an input.
You remove mass by having some boundary condition that removes mass -- like an exhaust vent.
For the warning please read the user's guide chapter on defining species.

**Reply by ESH-y** [reply]
1. at the time the simulation crush the only vent working is the fire.
this is my definition for fire:

&SPEC ID='POLYURETHANE_REAC_fuel', FORMULA='C6.3H7.1O2.1N1.0'/

&REAC ID='POLYURETHANE_REAC',
      FYI='NFPA Babrauskas',
      FUEL='POLYURETHANE_REAC_fuel',
      CO_YIELD=0.0464,
      SOOT_YIELD=0.1664/


&SURF ID='burner car EL',
      COLOR='RED',
      HRRPUA=590.0,
      RAMP_Q='burner car EL_RAMP_Q',
      E_COEFFICIENT=0.015/
&RAMP ID='burner car EL_RAMP_Q', T=0.0, F=0.0/
&RAMP ID='burner car EL_RAMP_Q', T=120.0, F=0.08/
&RAMP ID='burner car EL_RAMP_Q', T=240.0, F=0.75/
&RAMP ID='burner car EL_RAMP_Q', T=350.0, F=1.0/
&RAMP ID='burner car EL_RAMP_Q', T=500.0, F=0.85/
&RAMP ID='burner car EL_RAMP_Q', T=700.0, F=0.51/
&RAMP ID='burner car EL_RAMP_Q', T=800.0, F=0.51/
&RAMP ID='burner car EL_RAMP_Q', T=900.0, F=0.85/
&RAMP ID='burner car EL_RAMP_Q', T=1200.0, F=0.34/
&RAMP ID='burner car EL_RAMP_Q', T=2000.0, F=0.0/

2. is it possible to add a line to the code to fix the density ?

thank you for your replay

**Comment by TristanHehnen** [comment]
Hi, are you using GEOM in your simulation setup or is all the geometry made of OBST? I encounter this message frequently when using GEOM.

**Comment by ESH-y** [comment]
Hi'
what is GEOM please?
im bringing model from REVIT to pyrosim. not all of the model is OBST, only the hvac parts that I'm building is OBST.

**Reply by TristanHehnen** [reply]
It is a feature in development to use unstructured geometries with FDS, see section ["7.3 Creating Unstructured Geometry (Beta)" in the User's Guide](https://github.com/firemodels/fds/releases/tag/FDS-6.8.0).

Do you voxelise the CAD model coming from REVIT in Pyrosim? 
Are you building the HVAC system as actual hollow ducts/pipes, or are you using OBST to attach SURF with the desired parameters?

**Comment by mcgratta** [comment]
If your simulation lasted only 11 s, it might mean that something is evolving rapidly in the simulation which drives the temperature up and the density down. This might be caused by a numerical instability and the message about minimum density is telling you that FDS is trying to recover. Often this message appears before a numerical instability is announced and the simulation is ended. I suggest that you determine where in the domain you see high temperatures and/or low densities using `TEMPERATURE` and `DENSITY` slices. Then review your assumptions and set up. Are you introducing mass at a rapid rate? Very high heat release rate? These are often the things that trigger an early end of the simulaiton.
